src/MyConnection.h

```diff
@@ -30,15 +35,21 @@ class MyConnection : boost::noncopyable {
     mysql_options(pConn_, MYSQL_OPT_LOCAL_INFILE, 0);
     // Default to UTF-8
     mysql_options(pConn_, MYSQL_SET_CHARSET_NAME, "UTF8");
-    if (groups != "")
-      mysql_options(pConn_, MYSQL_READ_DEFAULT_GROUP, groups.c_str());
-    if (default_file != "")
-      mysql_options(pConn_, MYSQL_READ_DEFAULT_FILE, default_file.c_str());
-
-
-    if (!mysql_real_connect(pConn_, host.c_str(), user.c_str(),
-        password.c_str(), db == "" ? NULL : db.c_str(), port,
-        unix_socket == "" ? NULL : unix_socket.c_str(), client_flag)) {
+    if (!groups.isNull())
+      mysql_options(pConn_, MYSQL_READ_DEFAULT_GROUP,
+                    Rcpp::as<std::string>(groups).c_str());
+    if (!default_file.isNull())
+      mysql_options(pConn_, MYSQL_READ_DEFAULT_FILE,
+                    Rcpp::as<std::string>(default_file).c_str());
+
+    if (!mysql_real_connect(pConn_,
+        host.isNull() ? NULL : Rcpp::as<std::string>(host).c_str(),
```

Comment:
Sorry it took so long. Just a minor nitpick: This pattern is repeated several times, I think it would help creating a small function here.